File: Lession 3/dfs.py
def dfs(graph,start):
    stack = [start]
    visited = []
    while stack:
        node = stack.pop()
        if node not in visited:
            visited.append(node)
            stack.extend(graph[node]-set(visited))
    return visited
graph = {'A': set(['B', 'C']),
         'B': set(['A', 'D', 'E']),
            'C': set(['A', 'F']),
            'D': set(['B']),
            'E': set(['B', 'F']),
            'F': set(['C', 'E'])}
dfs(graph, 'A')
         

File: Lession 3/dfs.jl
function dfs(graph, start)
    visited = Dict()
    for node in keys(graph)
        visited[node] = false
    end
    dfs_helper(graph, start, visited)
end
function main()
    graph = Dict()
    graph["A"] = ["B", "C"]
    graph["B"] = ["A", "D", "E"]
    graph["C"] = ["A", "F"]
    graph["D"] = ["B"]
    graph["E"] = ["B", "F"]
    graph["F"] = ["C", "E"]
    dfs(graph, "A")
end